applications/FluidDynamicsApplication/custom_processes/apply_mach_dependent_boundary_conditions.h

```diff
@@ -133,7 +133,7 @@ namespace Kratos
             /** @brief This object points to the proper (Fix|Free)Dof function
              * according to the time interval.
              */
-            std::function<decltype(FixDof)> mEnforceInternal = FreeDof;
+            decltype(FixDof) * mEnforceInternal = & FreeDof;
```

Comment:
Apparently function pointers are considerably faster than std::function. Since I know I won't bind a lambda, I decided to switch to function pointer.

This function is assigned once per time-step and variable, and called once per time-step, variable, and node, so the gains may accumulate.

A ran a quick benchmark of a simplified version and apparently std::function is between 5 to 6 times slower.
![OmBeMOJfzSKViWK9FIG2IRMdxpc](https://user-images.githubusercontent.com/47142856/135069435-e2b930cb-75b6-414b-a97e-dffcae41c411.png)

The benchmark: https://quick-bench.com/q/OmBeMOJfzSKViWK9FIG2IRMdxpc